src/EnergyPlus/Coils/CoilCoolingDX.cc

```diff
@@ -665,6 +665,8 @@ void CoilCoolingDX::simulate(EnergyPlusData &state,
                              bool const singleMode,
                              Real64 LoadSHR)
 {
+    assert(speedNum != 0);
```

Comment:
If the parent object has no load, or if set point control does not need to turn on, what is the speed set to? I would think it would be 0. But maybe not, it may be speed=1 and speedRatio = 0 (but that doesn't seem intuitive). The point is to allow a call to this sim function with the coil off so that the inlet condition can be passed to the outlet, even if the inlet mass flow = 0 or constant fan is active and the inlet mass flow > 0. If you pass inlet to outlet for a parent that is off, you also have to pass inlet to outlet for all the children. When a user looks at the cooling coil speed report and they know that the coil is off (e.g., winter) and see speed = 1 they would likely think the coil is on. If speed = 1 and speedRatio = 0 means the coil is off then this change is fine. I just think that if the coil is off then speed = 0, cyclingRatio = 0 and speedRatio = 0. One or more of those variables are only non-zero if the coil is on.

Reply:
@rraustad, I believe that we've established that there are no use cases where speed=0 is used intentionally except as you outlined (below). The very few regression failures that are left are happening because a handful of simulation paths go through branches where coil speed in a multispeed coil is just not set, which would be a separate issue to log. Could we address this PR with this in mind?

> If the parent object has no load, or if set point control does not need to turn on, what is the speed set to? I would think it would be 0. But maybe not, it may be speed=1 and speedRatio = 0 (but that doesn't seem intuitive). The point is to allow a call to this sim function with the coil off so that the inlet condition can be passed to the outlet, even if the inlet mass flow = 0 or constant fan is active and the inlet mass flow > 0. If you pass inlet to outlet for a parent that is off, you also have to pass inlet to outlet for all the children. When a user looks at the cooling coil speed report and they know that the coil is off (e.g., winter) and see speed = 1 they would likely think the coil is on. If speed = 1 and speedRatio = 0 means the coil is off then this change is fine. I just think that if the coil is off then speed = 0, cyclingRatio = 0 and speedRatio = 0. One or more of those variables are only non-zero if the coil is on.